lib/brakeman/tracker/collection.rb

```diff
@@ -56,7 +56,9 @@ def add_method visibility, name, src, file_name
         @class_methods[name] = meth_info
 
         # TODO fix this weirdness
-        name = :"#{src[1]}.#{name}"
+        # See https://github.com/presidentbeef/brakeman/pull/881 for context
+        prefix = sexp?(src[1]) ? src[1][0] : src[1]
+        name = :"#{prefix}.#{name}"
```

Comment:
I think this won't work if `src[1]` is not `self`.

E.g.

```ruby
def A.foo
  # ...
end
```

`src[1].value` might work better (I haven't tested thoroughly).